core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -211,7 +211,7 @@ class BrokerToControllerChannelManagerImpl(
         50,
         Selectable.USE_DEFAULT_BUFFER_SIZE,
         Selectable.USE_DEFAULT_BUFFER_SIZE,
-        config.requestTimeoutMs,
+        Math.min(Int.MaxValue, Math.min(config.controllerSocketTimeoutMs, retryTimeoutMs)).toInt, // request timeout should not exceed the provided retry timeout
```

Comment:
let's just use `retryTimeoutMs`, there is no need to involve `controllerSocketTimeoutMs`. That is not a KRaft config anyway...

Reply:
I don't think we can use the retry timeout directly.  AlterPartitionManager sets [retryTimeoutMs = Long.MaxValue](https://github.com/apache/kafka/blob/42cfd57a2498b1a997e77cc24ec38b3722cc1f6a/core/src/main/scala/kafka/server/AlterPartitionManager.scala#L94).  I believe we have to set a reasonable request timeout -- something that doesn't exceed the retry timeout, but also something that isn't effectively forever.